cli/reactjs_jsx_ppx_v3.ml

```diff
@@ -162,77 +170,114 @@ let makeModuleName fileName nestedModules fnName =
   constructor and a props external
 *)
 
-(* Build an AST node representing all named args for the `external` definition for a component's props *)
-let rec recursivelyMakeNamedArgsForExternal list args =
-  match list with
-  | (label, default, loc, interiorType) :: tl ->
-      recursivelyMakeNamedArgsForExternal tl
-        (Typ.arrow ~loc label
-           ( match (label, interiorType, default) with
-           (* ~foo=1 *)
-           | label, None, Some _ ->
-               { ptyp_desc = Ptyp_var (safeTypeFromValue label); ptyp_loc = loc; ptyp_attributes = [] }
-           (* ~foo: int=1 *)
-           | _label, Some type_, Some _ -> type_
-           (* ~foo: option(int)=? *)
-           | label, Some { ptyp_desc = Ptyp_constr ({ txt = Lident "option" }, [ type_ ]) }, _
-           | label, Some { ptyp_desc = Ptyp_constr ({ txt = Ldot (Lident "*predef*", "option") }, [ type_ ]) }, _
-           (* ~foo: int=? - note this isnt valid. but we want to get a type error *)
-           | label, Some type_, _
-             when isOptional label ->
-               type_
-           (* ~foo=? *)
-           | label, None, _ when isOptional label ->
-               { ptyp_desc = Ptyp_var (safeTypeFromValue label); ptyp_loc = loc; ptyp_attributes = [] }
-           (* ~foo *)
-           | label, None, _ -> { ptyp_desc = Ptyp_var (safeTypeFromValue label); ptyp_loc = loc; ptyp_attributes = [] }
-           | _label, Some type_, _ -> type_ )
-           args)
-  | [] -> args
-  [@@raises Invalid_argument]
-
-(* Build an AST node for the [@bs.obj] representing props for a component *)
-let makePropsValue fnName loc namedArgListWithKeyAndRef propsType =
-  let propsName = fnName ^ "Props" in
-  {
-    pval_name = { txt = propsName; loc };
-    pval_type =
-      recursivelyMakeNamedArgsForExternal namedArgListWithKeyAndRef
-        (Typ.arrow nolabel
-           { ptyp_desc = Ptyp_constr ({ txt = Lident "unit"; loc }, []); ptyp_loc = loc; ptyp_attributes = [] }
-           propsType);
-    pval_prim = [ "" ];
-    pval_attributes = [ ({ txt = "bs.obj"; loc }, PStr []) ];
-    pval_loc = loc;
-  }
-  [@@raises Invalid_argument]
-
-(* Build an AST node representing an `external` with the definition of the [@bs.obj] *)
-let makePropsExternal fnName loc namedArgListWithKeyAndRef propsType =
-  { pstr_loc = loc; pstr_desc = Pstr_primitive (makePropsValue fnName loc namedArgListWithKeyAndRef propsType) }
-  [@@raises Invalid_argument]
-
-(* Build an AST node for the signature of the `external` definition *)
-let makePropsExternalSig fnName loc namedArgListWithKeyAndRef propsType =
-  { psig_loc = loc; psig_desc = Psig_value (makePropsValue fnName loc namedArgListWithKeyAndRef propsType) }
-  [@@raises Invalid_argument]
+(* List.filter_map in 4.08.0 *)
+let filterMap f =
```

Comment:
This is not needed anymore after rebasing on master.

Reply:
Oh! nice.